controllers/argocd_metrics_controller.go

```diff
@@ -109,73 +109,128 @@ func (r *ArgoCDMetricsReconciler) Reconcile(ctx context.Context, request reconci
 	}
 
 	const clusterMonitoringLabel = "openshift.io/cluster-monitoring"
-	_, exists := namespace.Labels[clusterMonitoringLabel]
-	if !exists {
-		if namespace.Labels == nil {
-			namespace.Labels = make(map[string]string)
+	labelVal, exists := namespace.Labels[clusterMonitoringLabel]
+
+	if argocd.Spec.Monitoring.Enabled {
```

Comment:
Just adding this as a note, no immediate action required.  [argocd.Spec.Monitoring.Enabled](https://github.com/argoproj-labs/argocd-operator/blob/4bd23cbd1a30f71eff97a4941b7485c914032e45/api/v1beta1/argocd_types.go#L687-L694) is used to control  creation of Prometheus alerting rules. With this change, we are reusing this field to enable/disable metric scraping, which makes sense to me. However, the current change won't support use-cases where user wants metric scraping enabled but alerting disabled OOTB. I am not sure how common this use-case is, but if we decide to support it in future, we could introduce a new field `argocd.Spec.Monitoring.Alerts` to control alerts rule creation.